src/lib/viz/legend/LegendWidget.ts

```diff
@@ -0,0 +1,57 @@
+import { Layer } from '@/viz/layer';
+import { queryDOMElement } from '@/core/utils/dom';
+import { CartoError } from '@/core/errors/CartoError';
+
+export type LegendGeometryType = 'point' | 'line' | 'polygon';
+
+export type LegendProperties = {
+  type: LegendGeometryType;
+  color?: string;
+  marker?: string;
+  strokeColor?: string;
+  strokeStyle?: string;
```

Comment:
Sorry, what's strokeStyle?

Reply:
A bit explanation about LegendProperties:
LegendProperties are the list of the properties that are supported by airship legend widgets. You can see these properties here https://carto.com/developers/airship/reference/#/components/legends?a=props
As you can see there is a strokeStyle property with these possible values https://developer.mozilla.org/en-US/docs/Web/CSS/border-style
Our styles helpers don't support this property but I think that can be interested If the user want to set it manually but we could remove it.

Reply:
Yeah, we should probably review them. We don't need to repeat all the previous props if we think that some don't make sense now.. Let's postpone the decission on this.